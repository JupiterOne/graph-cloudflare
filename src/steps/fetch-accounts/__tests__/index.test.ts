import {
  executeStepWithDependencies,
  Recording,
  setupRecording,
} from '@jupiterone/integration-sdk-testing';
import { buildStepTestConfigForStep } from '../../../../test/config';

let recording: Recording;

afterEach(async () => {
  await recording.stop();
});

test('should process account entities', async () => {
  recording = setupRecording({
    name: 'cloudflare_account',
    directory: __dirname,
    redactedRequestHeaders: ['api-token', 'X-Auth-Key', 'X-Auth-Email'],
    options: {
      recordFailedRequests: false,
      matchRequestsBy: {
        url: {
          query: false,
        },
      },
    },
  });

  const stepConfig = buildStepTestConfigForStep('fetch-accounts');
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
});
