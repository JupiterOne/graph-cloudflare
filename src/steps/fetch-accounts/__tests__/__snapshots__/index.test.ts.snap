// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should process account entities 1`] = `
Array [
  Object {
    "_class": Array [
      "Account",
    ],
    "_key": "cloudflare_account:cd12fd3d7190eff1985f713c47015663",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "created_on": "2023-02-10T17:42:13.428084Z",
          "id": "cd12fd3d7190eff1985f713c47015663",
          "legacy_flags": Object {
            "enterprise_zone_quota": Object {
              "available": 0,
              "current": 0,
              "maximum": 0,
            },
          },
          "name": "David.obregon@contractor.jupiterone.com's Account",
          "settings": Object {
            "access_approval_expiry": null,
            "api_access_enabled": null,
            "enforce_twofactor": false,
            "use_account_custom_ns_by_default": false,
          },
          "type": "standard",
        },
      },
    ],
    "_type": "cloudflare_account",
    "accessApprovalExpiry": null,
    "accountId": "cd12fd3d7190eff1985f713c47015663",
    "createdOn": 1676050933428,
    "displayName": "David.obregon@contractor.jupiterone.com's Account",
    "id": "cd12fd3d7190eff1985f713c47015663",
    "mfaEnabled": undefined,
    "mfaEnforced": false,
    "name": "David.obregon@contractor.jupiterone.com's Account",
    "type": "standard",
  },
  Object {
    "_class": Array [
      "AccessRole",
    ],
    "_key": "cloudflare_account_role:05784afa30c1afe1440e79d9351c7430",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "description": "Can access the full account, except for membership management and billing.",
          "id": "05784afa30c1afe1440e79d9351c7430",
          "name": "Administrator",
          "permissions": Object {
            "access": Object {
              "edit": true,
              "read": true,
            },
            "analytics": Object {
              "edit": false,
              "read": true,
            },
            "app": Object {
              "edit": true,
              "read": false,
            },
            "auditlogs": Object {
              "edit": false,
              "read": true,
            },
            "billing": Object {
              "edit": false,
              "read": true,
            },
            "blocks": Object {
              "edit": true,
              "read": true,
            },
            "cache_purge": Object {
              "edit": true,
              "read": false,
            },
            "dns_records": Object {
              "edit": true,
              "read": true,
            },
            "fbm": Object {
              "edit": true,
              "read": true,
            },
            "fbm_acc": Object {
              "edit": true,
              "read": false,
            },
            "healthchecks": Object {
              "edit": true,
              "read": true,
            },
            "http_applications": Object {
              "edit": true,
              "read": true,
            },
            "image": Object {
              "edit": true,
              "read": true,
            },
            "lb": Object {
              "edit": true,
              "read": true,
            },
            "legal": Object {
              "edit": false,
              "read": true,
            },
            "logs": Object {
              "edit": true,
              "read": true,
            },
            "magic": Object {
              "edit": true,
              "read": true,
            },
            "member": Object {
              "edit": false,
              "read": true,
            },
            "organization": Object {
              "edit": true,
              "read": true,
            },
            "ssl": Object {
              "edit": true,
              "read": true,
            },
            "stream": Object {
              "edit": true,
              "read": true,
            },
            "subscription": Object {
              "edit": true,
              "read": true,
            },
            "teams": Object {
              "edit": true,
              "read": true,
              "report": true,
            },
            "waf": Object {
              "edit": true,
              "read": true,
            },
            "waitingroom": Object {
              "edit": true,
              "read": true,
            },
            "webhooks": Object {
              "edit": true,
              "read": true,
            },
            "worker": Object {
              "edit": true,
              "read": true,
            },
            "zaraz": Object {
              "edit": true,
              "publish": true,
              "read": true,
            },
            "zone": Object {
              "edit": true,
              "read": true,
            },
            "zone_settings": Object {
              "edit": true,
              "read": true,
            },
            "zone_versioning": Object {
              "edit": true,
              "read": true,
            },
          },
        },
      },
    ],
    "_type": "cloudflare_account_role",
    "createdOn": undefined,
    "description": "Can access the full account, except for membership management and billing.",
    "displayName": "Administrator",
    "id": "05784afa30c1afe1440e79d9351c7430",
    "name": "Administrator",
    "roleId": "05784afa30c1afe1440e79d9351c7430",
  },
  Object {
    "_class": Array [
      "User",
    ],
    "_key": "cloudflare_account_member:e265512b21d07f6cf2c355a1a67d172f",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "api_access_enabled": null,
          "id": "e265512b21d07f6cf2c355a1a67d172f",
          "roles": Array [
            Object {
              "description": "Can edit any Cloudflare setting, make purchases, update billing, and manage memberships. Super Administrators can revoke the access of other Super Administrators.",
              "id": "33666b9c79b9a5273fc7344ff42f953d",
              "name": "Super Administrator - All Privileges",
              "permissions": Object {
                "access": Object {
                  "edit": true,
                  "read": true,
                },
                "analytics": Object {
                  "edit": false,
                  "read": true,
                },
                "app": Object {
                  "edit": true,
                  "read": false,
                },
                "auditlogs": Object {
                  "edit": false,
                  "read": true,
                },
                "billing": Object {
                  "edit": true,
                  "read": true,
                },
                "blocks": Object {
                  "edit": true,
                  "read": true,
                },
                "cache_purge": Object {
                  "edit": true,
                  "read": false,
                },
                "dash_sso": Object {
                  "edit": true,
                  "read": true,
                },
                "dns_records": Object {
                  "edit": true,
                  "read": true,
                },
                "fbm": Object {
                  "edit": true,
                  "read": true,
                },
                "fbm_acc": Object {
                  "edit": true,
                  "read": false,
                },
                "healthchecks": Object {
                  "edit": true,
                  "read": true,
                },
                "http_applications": Object {
                  "edit": true,
                  "read": true,
                },
                "image": Object {
                  "edit": true,
                  "read": true,
                },
                "lb": Object {
                  "edit": true,
                  "read": true,
                },
                "legal": Object {
                  "edit": true,
                  "read": true,
                },
                "logs": Object {
                  "edit": true,
                  "read": true,
                },
                "magic": Object {
                  "edit": true,
                  "read": true,
                },
                "member": Object {
                  "edit": true,
                  "read": true,
                },
                "organization": Object {
                  "edit": true,
                  "read": true,
                },
                "ssl": Object {
                  "edit": true,
                  "read": true,
                },
                "stream": Object {
                  "edit": true,
                  "read": true,
                },
                "subscription": Object {
                  "edit": true,
                  "read": true,
                },
                "teams": Object {
                  "edit": true,
                  "pii": true,
                  "read": true,
                  "report": true,
                },
                "waf": Object {
                  "edit": true,
                  "read": true,
                },
                "waitingroom": Object {
                  "edit": true,
                  "read": true,
                },
                "webhooks": Object {
                  "edit": true,
                  "read": true,
                },
                "worker": Object {
                  "edit": true,
                  "read": true,
                },
                "zaraz": Object {
                  "edit": true,
                  "publish": true,
                  "read": true,
                },
                "zone": Object {
                  "edit": true,
                  "read": true,
                },
                "zone_settings": Object {
                  "edit": true,
                  "read": true,
                },
                "zone_versioning": Object {
                  "edit": true,
                  "read": true,
                },
              },
            },
          ],
          "status": "accepted",
          "user": Object {
            "email": "david.obregon@contractor.jupiterone.com",
            "first_name": null,
            "id": "6cb57293541555d251de7f6a1f490420",
            "last_name": null,
            "two_factor_authentication_enabled": false,
          },
        },
      },
    ],
    "_type": "cloudflare_account_member",
    "active": true,
    "admin": true,
    "createdOn": undefined,
    "displayName": "david.obregon@contractor.jupiterone.com",
    "email": "david.obregon@contractor.jupiterone.com",
    "firstName": null,
    "id": "6cb57293541555d251de7f6a1f490420",
    "lastName": null,
    "membershipId": "e265512b21d07f6cf2c355a1a67d172f",
    "mfaEnabled": false,
    "name": "",
    "roles": Array [
      "33666b9c79b9a5273fc7344ff42f953d",
    ],
    "status": "accepted",
    "superAdmin": true,
    "userId": "6cb57293541555d251de7f6a1f490420",
    "username": "david.obregon@contractor.jupiterone.com",
  },
]
`;

exports[`should process account entities 2`] = `
Array [
  Object {
    "_class": "HAS",
    "_fromEntityKey": "cloudflare_account:cd12fd3d7190eff1985f713c47015663",
    "_key": "cloudflare_account:cd12fd3d7190eff1985f713c47015663|has|cloudflare_account_role:05784afa30c1afe1440e79d9351c7430",
    "_toEntityKey": "cloudflare_account_role:05784afa30c1afe1440e79d9351c7430",
    "_type": "cloudflare_account_has_role",
    "displayName": "HAS",
  },
  Object {
    "_class": "HAS",
    "_fromEntityKey": "cloudflare_account:cd12fd3d7190eff1985f713c47015663",
    "_key": "cloudflare_account:cd12fd3d7190eff1985f713c47015663|has|cloudflare_account_member:e265512b21d07f6cf2c355a1a67d172f",
    "_toEntityKey": "cloudflare_account_member:e265512b21d07f6cf2c355a1a67d172f",
    "_type": "cloudflare_account_has_member",
    "displayName": "HAS",
  },
  Object {
    "_class": "ASSIGNED",
    "_fromEntityKey": "cloudflare_account_member:e265512b21d07f6cf2c355a1a67d172f",
    "_key": "cloudflare_account_member:e265512b21d07f6cf2c355a1a67d172f|assigned|cloudflare_account_role:33666b9c79b9a5273fc7344ff42f953d",
    "_toEntityKey": "cloudflare_account_role:33666b9c79b9a5273fc7344ff42f953d",
    "_type": "cloudflare_account_member_assigned_role",
    "displayName": "ASSIGNED",
  },
]
`;
